src/lambda/models/samLambdaRuntime.ts

```diff
@@ -3,76 +3,61 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { Runtime } from 'aws-sdk/clients/lambda'
 import * as immutable from 'immutable'
 
-// TODO: Can we dynamically determine the available runtimes? We could theoretically parse the
-// output of `sam init --help`, but that's a hack.
-export type SamLambdaRuntime = 'python3.7' | 'python3.6' | 'python2.7' | 'nodejs8.10' | 'nodejs10.x' | 'dotnetcore2.1'
+export const nodeJsRuntimes: Set<Runtime> = new Set<Runtime>(['nodejs8.10', 'nodejs10.x'])
```

Comment:
Seeing `samLambdaRuntimes` with immutable Set makes me wonder if these sets should also be immutable

Reply:
Made them immutable, including the map below